instrumentation/testing/testing.go

```diff
@@ -242,21 +238,33 @@ func (test *Test) end() {
 	test.span.FinishWithOptions(finishOptions)
 }
 
-func (test *Test) extractTestLoggerOutput() {
+func (test *Test) extractTestLoggerOutputWithFailOrSkipReason(failedOrSkipped bool) (reason, source string) {
```

Comment:
Why do you need to pass in `failedOrSkipped`? You have access to the test in this method.

Also, I don't think we need to return anything here. The last event in the case of a test failed or skipped should have a different event type. So we can contain all the required logic here.

Reply:
Agree with the parameters.

So do you prefer to write the failed and skipped event here in this func, and remove it from the `end()` func?  Because right now I have only the logs in this func and the other events in the end func. That's the reason to the return in this method, to avoid accessing the log buffer twice.

Reply:
I think it’s simpler if we do it here

Reply:
done